src/library/scala/concurrent/BatchingExecutor.scala

```diff
@@ -26,7 +26,13 @@ trait Batchable {
 
 private[concurrent] object BatchingExecutorStatics {
   final val emptyBatchArray: Array[Runnable] = new Array[Runnable](0)
-  final val marker = ""
+
+  // Max number of Runnables executed nested before starting to batch (to prevent stack exhaustion)
+  final val syncPreBatchDepth = 16
```

Comment:
16 seems like a sweet-spot in terms of tradeoffs, but who knows?

Reply:
I contemplated having it as a parameter to submitSyncBatched, but it would be weird if there are multiple invocations which use different numbers in the same implementation. And putting it as a method on a trait means an invokeinterface extra per call, which is costly. Having it as a JVM-parameter means that it is set for all implementations which may or may not be desirable either. I'm not sure what the best option is in this case, but it is a good question—do you have any preference and why?